pkg/gui/controllers/branches_controller.go

```diff
@@ -157,36 +180,30 @@ func (self *BranchesController) setUpstream(selectedBranch *models.Branch) error
 						return self.c.Error(err)
 					}
 					return nil
-				},
-				Key: 'u',
+				})
 			},
-			{
-				LabelColumns: []string{self.c.Tr.SetUpstream},
-				OnPress: func() error {
-					return self.c.Helpers().Upstream.PromptForUpstreamWithoutInitialContent(selectedBranch, func(upstream string) error {
-						upstreamRemote, upstreamBranch, err := self.c.Helpers().Upstream.ParseUpstream(upstream)
-						if err != nil {
-							return self.c.Error(err)
-						}
+			Key: 's',
+		},
+	}
 
-						if err := self.c.Git().Branch.SetUpstream(upstreamRemote, upstreamBranch, selectedBranch.Name); err != nil {
-							return self.c.Error(err)
-						}
-						if err := self.c.Refresh(types.RefreshOptions{
-							Mode: types.SYNC,
-							Scope: []types.RefreshableView{
-								types.BRANCHES,
-								types.COMMITS,
-							},
-						}); err != nil {
-							return self.c.Error(err)
-						}
-						return nil
-					})
-				},
-				Key: 's',
+	if selectedBranch.IsTrackingRemote() {
+		options = append(options, &types.MenuItem{
+			LabelColumns: []string{self.c.Tr.ViewUpstreamResetOptions},
+			OpensMenu:    true,
+			OnPress: func() error {
+				err := self.c.Helpers().Refs.CreateGitResetMenu(fmt.Sprintf("%s/%s", selectedBranch.UpstreamRemote, selectedBranch.Name))
+				if err != nil {
+					return self.c.Error(err)
+				}
+				return nil
 			},
-		},
+			Key: 'g',
+		})
+	}
+
+	return self.c.Menu(types.CreateMenuOptions{
+		Title: self.c.Tr.Actions.SetUnsetUpstream,
```

Comment:
Now that we're doing more than just setting/unsetting the upstream in this menu, can we:
* change the menu title to be 'Upstream options'
* update the keybinding to have the label 'View upstream options'
* Give the keybinding a tooltip saying 'View options relating to the branch's upstream e.g. setting/unsetting the upstream and resetting to the upstream'